doc/languages-frameworks/rust.section.md

```diff
@@ -368,6 +368,15 @@ rustPlatform.buildRustPackage rec {
 }
 ```
 
+## Compiling packages that use the `bindgen` crate {#compiling-with-rust-bindgen}
+
+`bindgen` is a crate that is used to generate ffi binding to C/C++ libraries. It can be used by upstream build systems either as an executable (available as `rust-bindgen`) or directly as a crate in `build.rs`.
+In the latter case, the build will fail while bindgen is looking for libclang.so:
+```
+thread 'main' panicked at 'Unable to find libclang: ...
+```
+To allow bindgen to find libclang and all the libraries in `buildInputs`, add `rustBindgenHook` to `nativeBuildInputs`.
```

Comment:
The rust docs have a section for `hooks`, could you add this there as well please?

Reply:
It's only for hooks un rustPlatform. But it wouldn't make sense to add this one to rustPlatform, as it does not depend on rust...

Reply:
> as it does not depend on rust...

I don't understand?

Reply:
TBH I think it would be somewhat confusing for it separate when it is mostly going to be used with `rustPlatform` anyway just to set some env vars and it doesn't depend on `pkgs.bindgen`?

Reply:
rust is not a dependency of this hook. only clang is. But I did it anyway.